src/main/java/org/zalando/riptide/handler/OAuth2CompatibilityResponseErrorHandler.java

```diff
@@ -0,0 +1,41 @@
+package org.zalando.riptide.handler;
+
+/*
+ * ⁣​
+ * riptide
+ * ⁣⁣
+ * Copyright (C) 2015 Zalando SE
+ * ⁣⁣
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ​⁣
+ */
+
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.zalando.riptide.AlreadyConsumedResponseException;
+
+import java.io.IOException;
+
+public final class OAuth2CompatibilityResponseErrorHandler implements ResponseErrorHandler {
```

Comment:
A class comment would help to understand the need for this class. It could incorporate the relevant code pieces from the OAuthResponseErrorHandler to clarify the issue it solves.